### PR TITLE
fix(tracing): Performance Tracing should be disabled by default

### PR DESCRIPTION
**Pull request by krystofwoldrich**
## :loudspeaker: Type of change

- [x] Bugfix

## :scroll: Description

- introduced by https://github.com/getsentry/sentry-react-native/pull/2933

## :green_heart: How did you test it?
unit tests, sample app

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

File: src/js/integrations/default.ts [hunk 1]
```diff
@@ -77,7 +76,11 @@ export function getDefaultIntegrations(options: ReactNativeClientOptions): Integ
     }
   }
 
-  if (hasTracingEnabled(options) && options.enableAutoPerformanceTracing) {
+  const hasTracingEnabled =
```
Comment (kahest) [inline src/js/integrations/default.ts:79]: `l`: maybe worth a comment why `hasTracingEnabled(options)` from core utils doesn't work

File: src/js/sdk.tsx [hunk 2]
```diff
@@ -76,8 +76,11 @@ export function init(passedOptions: ReactNativeOptions): void {
     stackParser: stackParserFromStackParserOptions(passedOptions.stackParser || defaultStackParser),
     beforeBreadcrumb: safeFactory(passedOptions.beforeBreadcrumb, { loggerMessage: 'The beforeBreadcrumb threw an error' }),
     initialScope: safeFactory(passedOptions.initialScope, { loggerMessage: 'The initialScope threw an error' }),
-    tracesSampler: safeTracesSampler(passedOptions.tracesSampler),
   };
+  if ('tracesSampler' in options) {
```
Comment (kahest) [inline src/js/sdk.tsx:80]: why is this necessary? also, the impl of `safeTracesSampler` looks strange to my untrained (in TS/JS) eyes, it's basically `if (tracesSampler) { ... } else { return tracesSampler; }`
probably totally fine, just trying to understand & learn :)

Comment (kahest) [reply src/js/sdk.tsx:80]: ok yeah that's what I assumed - and we want to make sure we don't set a falsy value and therefore guard it with the added `if`?

Comment (krystofwoldrich) [reply src/js/sdk.tsx:80]: Yes, because if we set undefined key is created on the options object which can cause issues with the key existence checks.

**Review by kahest — APPROVED** [review]
LGTM! 🚀
